Sources/Core/Permission.swift

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+//
+// You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
+// copy, modify, and distribute this software in source code or binary form for use
+// in connection with the web services and APIs provided by Facebook.
+//
+// As with any software that integrates with the Facebook platform, your use of
+// this software is subject to the Facebook Developer Principles and Policies
+// [http://developers.facebook.com/policy/]. This copyright notice shall be
+// included in all copies or substantial portions of the software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Foundation
+
+// TODO: Introduce a type to JSON encode permissions This object will be responsible for translating between JSON with snake_case keys and our canonical camelCased cases
+
+enum Permission: Hashable {
+  case `default`
```

Comment:
There's quite a few others: `publish_videos`, `pages_messaging`, all the various ones for Instagram and business permissions. Should we consider namespacing some of them (e.g. `Permission.Instagram.manageComments`)?

https://developers.facebook.com/docs/facebook-login/permissions#reference-default

Reply:
Yes and no. I had a few thoughts. 

1. I was trying to introduce the minimum at this point based on the yagni principle.
2. I have some vague idea that we'll be able to use framework name-spacing to accomplish this once we know where different permissions belong but I don't have this 100% fleshed out yet.

TLDR; I wanted to defer the decision of where to put the 'non-essential' permissions for as long as possible. Unless they all for sure belong in CoreKit in which case I'm happy to add them. :)